src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -185,13 +169,21 @@ private BitSet getFilteredDocsBitSet(final LeafReaderContext ctx) throws IOExcep
         return createBitSet(scorer.iterator(), liveDocs, maxDoc);
     }
 
-    private BitSet createBitSet(final DocIdSetIterator filteredDocIdsIterator, final Bits liveDocs, int maxDoc) throws IOException {
-        if (liveDocs == null && filteredDocIdsIterator instanceof BitSetIterator) {
+    private BitSet getAllDocsBitSet(final LeafReaderContext ctx) throws IOException {
+        final FloatVectorValues floatVectorValues = ctx.reader().getFloatVectorValues(this.knnQuery.getField());
+        final Bits liveDocs = ctx.reader().getLiveDocs();
+        final int maxDoc = ctx.reader().maxDoc();
+        final VectorScorer scorer = floatVectorValues.scorer(this.knnQuery.getQueryVector());
```

Comment:
this is pretty weird, since all we need is the iterator, you can get the iterator from KNNVectorValues too.

Reply:
Updated iterator to accept non filter use cases. We don't need this anymore